### ISSUE TITLE
Using Posts module with Themer: provide context-aware variable or checkbox for Custom Query

### ISSUE THREAD

**Issue by jas8522** [beaverbuilder/feature-requests#64]
THE GOAL:

I'm building a post category/archive page with themer. Above the default main query, I want to put a “Most Popular Posts” grid (containing posts from the visible/current category). The views data will come from post meta using WP Popular Posts.

In order to have this work, I need to both pull from the Main Query AND allow a couple query var differences:

1. Limiting of posts (I only want 3-6 of them), and
2. Sort by the greatest number of views to least number of views

However, when the Main Query is selected, there's no option to apply either of those changes, and when Custom Query is selected there's no option to tell that I only want posts from the current/contextual category.

I was able to accomplish what I wanted by using a posts module with Custom Query and applying the above query changes (sorting and limiting). I then overrided the query args via functions.php as [described here in the BB Docs](https://docs.wpbeaverbuilder.com/beaver-builder/developer/tutorials-guides/create-a-filter-to-customize-the-display-of-post-data/?fbclid=IwAR3XRmyrXFP3IMjJZlsfLbV0W3273h03-do2N_LY2VLZlcDjWqrHvP8_0F0) and using this query adjustment:

```
$categories = get_the_category(); 
$query_args['category__in'] = $categories[0]->term_id; 
```

While this works fine, I would think a cleaner implementation would allow me to do this via the UI entirely. Here's two ways I can see it working well:

1. When using themer with the posts module and a custom query, themer would provide a variable like %THIS% or %CURRENT% that you could place under filter tags or categories. This would apply a similar filter as my query args changes above, or
2. When themer is active and you're adding a posts module Custom Query to the page there could be an option that's simply a toggle like: "Only include posts in the currently active context"  perhaps with a description tooltip like "This starts the custom query with the posts from the main query."

**Comment by quasel** [comment]
Yeah some sort of "Query Builder" based on the main query would be nice - but that's a though topic as there could be many moving parts…

**Comment by Pross** [comment]
Unfortunately during pre_get_posts modules are not available .. it's too early in wp actions
